src/mesh/exodusII_io.C

```diff
@@ -286,7 +286,40 @@ void ExodusII_IO::read (const std::string & fname)
 
           // Assign extra integer IDs
           for (auto & id : extra_ids)
-            elem->set_extra_integer(id, std::round(elem_ids[id][elem->id()]));
+          {
+            const Real v = elem_ids[id][elem->id()];
+
+            if (v == Real(-1))
+              {
+                elem->set_extra_integer(id, DofObject::invalid_id);
+                continue;
+              }
+
+            const long long iv = std::llround(v);
```

Comment:
I noticed that we don't explicitly
```
#include <cmath>
```
in exodusII_io.C. Presumably it is already included implicitly somehow or another, but might be good to make it explicit as well.

Reply:
Possibly we should use dof_id_type here because `v` is supposed to be generated from it.

Reply:
Actually, uint_64 or unsigned long long because dof_id_type could be in 32-bits.

Reply:
Using dof_id_type here is tricky, because there's no `std::dof_id_type_round()` and no functions that overload on return type; we'd need some `#if LIBMESH_DOF_ID_BYTES > 4` and `#else` and such to use different functions in different cases.

But more to the point, it shouldn't be necessary; you're right that we'll hit other guards (an FP exception thrown if we're out of the long long range, a libmesh C++ exception thrown if we're in `long long` range but out of the 1-to-1 representation range, and then we have a redundant assert (via `cast_int` at the end) if somehow we made it past those but we're not in `dof_id_type` range.

Reply:
Will `std::llround` throw an error if `v` is between `2^63` to `2^64`?

Reply:
From the man page:
```
       Domain error: x is a NaN or infinite, or the rounded value is too large
              An invalid floating-point exception (FE_INVALID) is raised.

       These functions do not set errno.
```